src/Workspaces/CSharp/Portable/CodeGeneration/NamedTypeGenerator.cs

```diff
@@ -198,9 +198,8 @@ private static MemberDeclarationSyntax RemoveAllMembers(MemberDeclarationSyntax
             TypeDeclarationSyntax typeDeclaration;
             if (namedType.IsRecord)
             {
-                typeDeclaration = namedType.TypeKind is TypeKind.Class
-                    ? SyntaxFactory.RecordDeclaration(SyntaxFactory.Token(SyntaxKind.RecordKeyword), namedType.Name.ToIdentifierToken())
-                    : SyntaxFactory.RecordStructDeclaration(SyntaxFactory.Token(SyntaxKind.RecordKeyword), namedType.Name.ToIdentifierToken());
+                var declarationKind = namedType.TypeKind is TypeKind.Class ? SyntaxKind.RecordDeclaration : SyntaxKind.RecordStructDeclaration;
+                typeDeclaration = SyntaxFactory.RecordDeclaration(declarationKind, SyntaxFactory.Token(SyntaxKind.RecordKeyword), namedType.Name.ToIdentifierToken());
```

Comment:
does RecordStructDeclaration cause SyntaxFactory.RecordDeclaration to create the `struct` token?  If not, we need to do that here.